lib/payment.js

```diff
@@ -57,11 +57,30 @@ class Payment {
      * @param {Address} sender - Senders address
      * @param {Address} recipient - Recipient address
      * @param {Wallet|NahmiiProvider} [walletOrProvider] - An optional Wallet or NahmiiProvider instance
+     * @param {String} [senderRef] - Optional uuid identifying the payment. Must be unique per sender wallet. Random if undefined.
+     *
+     * @example
+     * // Normal, random sender reference
+     * const senderWallet = new nahmii.Wallet(...);
+     * const recipientWallet = new nahmii.Wallet(...);
+     * const paymentAmount = nahmii.MonetaryAmount.from(...);
+     * const payment = new nahmii.Payment(paymentAmount, senderWallet.address, recipientWallet.address, senderWallet);
+     *
+     * // Advanced, semantic sender reference. See: https://github.com/uuidjs/uuid
+     * const uuidNamespace = '706ac453-2691-41af-9fde-ac5f787da1ec';
+     * const paymentSubject = '...';
+     * const senderRef = uuidv5(paymentSubject, uuidNamespace);
+     * const payment = new nahmii.Payment(paymentAmount, senderWallet.address, recipientWallet.address, senderWallet, senderRef);
      */
-    constructor(amount, sender, recipient, walletOrProvider) {
+    constructor(amount, sender, recipient, walletOrProvider, senderRef) {
         if (!(amount instanceof MonetaryAmount))
             throw new TypeError('amount is not an instance of MonetaryAmount');
 
+        senderRef = senderRef || uuidv4();
+
+        if (!/^[a-f0-9]{8}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{12}$/i.test(senderRef))
+            throw new TypeError('senderRef is not an uuid string');
```

Comment:
I am exercising extreme pedantism here by suggesting the replacement of "an uuid" by "a uuid", os possibly better "a UUID". :)